src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -3935,6 +3941,77 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        _DeploymentSetClickOnceVersions
+
+    Sets ClickOnce versions
+    ============================================================
+    -->
+  <Target
+      Name="_DeploymentSetClickOnceVersions"
+      Condition="'$(GenerateClickOnceManifests)'=='true'">
+
+    <PropertyGroup>
+      <_DeploymentManifestTargetFrameworkMoniker>$(TargetFrameworkMoniker)</_DeploymentManifestTargetFrameworkMoniker>
+      <_DeploymentManifestTargetFrameworkVersion>$(TargetFrameworkVersion)</_DeploymentManifestTargetFrameworkVersion>
```

Comment:
Why this indirection rather than just passing it? (genuine question, not necessarily asking you to change this)

Reply:
> Why this indirection rather than just passing it? (genuine question, not necessarily asking you to change this)

TargetFrameworkMoniker and TargetFrameworkVersion are project-specific values - for Core projects these will match Core target. ClickOnce manifest can only target .NET FX. For Core projects we add Launcher and set _DeploymentManifestTargetFrameworkMoniker and _DeploymentManifestTargetFrameworkVersion to match Launcher executable target.

These properties are used to initialize TargetFrameworkVersion and TargetFrameworkMoniker parameters of GenerateDeploymentManifest task.